core/qt-ble.cpp

```diff
@@ -185,15 +190,27 @@ dc_status_t BLEObject::read(void *data, size_t size, size_t *actual)
 	if (receivedPackets.isEmpty())
 		return DC_STATUS_IO;
 
-	QByteArray packet = receivedPackets.takeFirst();
+	int offset = 0;
+	while (!receivedPackets.isEmpty()) {
+		while (!receivedPackets.isEmpty()) {
```

Comment:
two nested loops with the same condition? That seems at least worth a comment, right?